### PR TITLE
Optimize docker image and update package

### PR DESCRIPTION
**Pull request by Gauvino**
## Checklist

- [X] I have read the [Contributing Guidelines]
- [X] I acknowledge that any submitted code will be licensed under the [ISC License]
- [X] I confirm that submitted code is my own work
- [X] I have tested the code, and confirm that it works

## Enviroment

- Operating System: WIN11 - Docker Desktop: v24.0.6
- Node version: 20.9.0
- npm version: 10.2.1

## Description

This PR update the repository to nodejs 20.9, npm 10.2.1 and all the sub package, adds .dockerignore file to remove form the built image the useless file and changed the node based image to alpine for having a really lightweight image and efficient one reducing the size overall by almost 1Gb (as you can see in the screenshot below)

(Don't forget to edit the descripition of the [dev/0.15.0] PR)

Tested the image on Docker Desktop (v24.0.6) on Windows (11) and on Ubuntu 23.10 (Mantic) (v24.0.6)

chrome_EdrkeE4K1E
chrome_t1csmbUX7Y
chrome_prRiz5bugT

Superseded: https://github.com/tycrek/ass/pull/246

[Contributing Guidelines]: https://github.com/tycrek/ass/blob/master/.github/CONTRIBUTING.md
[ISC License]: https://github.com/tycrek/ass/blob/master/LICENSE

### CODE REVIEW

**Comment by tycrek** [comment]
**[AUTOMATED MESSAGE]** 🔴 This Pull Request has merge conflicts. Please resolve the conflicts so the PR can be successfully reviewed and merged.

**Comment by tycrek** [comment]
**[AUTOMATED MESSAGE]** 🟢 Conflicts have been resolved.

**Comment by tycrek** [comment]
**[AUTOMATED MESSAGE]** 🔴 This Pull Request has merge conflicts. Please resolve the conflicts so the PR can be successfully reviewed and merged.

**Comment by tycrek** [comment]
**[AUTOMATED MESSAGE]** 🟢 Conflicts have been resolved.

**Comment by Gauvino** [comment]
Go look on discord before merging ! @tycrek

File: package-lock.json
Comment (tycrek) [inline package-lock.json:1]: Can you re-pull my merge, run `npm i`, and push any changes? Not sure why this file still has a diff

**Review by tycrek — APPROVED** [review]

**Pull request merged**